timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/api/NamespaceMatcher.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.api;
+
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+
+public interface NamespaceMatcher {
+    NamespaceMatcher NEVER_MATCH = ignored -> false;
+    boolean matches(Namespace namespace);
```

Comment:
As mentioned earlier, I really don't think we should use the `keyvalue.api.Namespace` here. A separate string-wrapper class makes sense though.